cmake/modules/SwiftSharedCMakeConfig.cmake

```diff
@@ -42,13 +42,33 @@ endfunction()
 
 function(get_imported_library_prefix outvar target prefix)
   string(FIND "${target}" "${prefix}" ALREADY_HAS_PREFIX)
-  if (ALREADY_HAS_PREFIX)
+  if (ALREADY_HAS_PREFIX EQUAL 0)
     set(${outvar} "" PARENT_SCOPE)
   else()
     set(${outvar} "${prefix}" PARENT_SCOPE)
   endif()
 endfunction()
 
+function(get_imported_library_path outvar target target_type)
+  if ("${target_type}" STREQUAL "EXECUTABLE")
+    set(${outvar} "${LLVM_BINARY_OUTPUT_INTDIR}/${target}" PARENT_SCOPE)
+    return()
+  endif()
+
+  # LLVM compiles module libraries as
+  # ${MODULE_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX} on both Linux and macOS.
```

Comment:
It looks like the correct variable is [CMAKE_SHARED_MODULE_SUFFIX](https://cmake.org/cmake/help/v3.0/variable/CMAKE_SHARED_MODULE_SUFFIX.html). Is there really no less ad hoc way to do this?